staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -595,8 +595,11 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 	// underlying watchCache is calling processEvent under its lock.
 	c.watchCache.RLock()
 	defer c.watchCache.RUnlock()
-	watchRV = startWatchResourceVersionFn()
-	cacheInterval, err := c.watchCache.getAllEventsSinceLocked(watchRV)
+	if err = c.waitUntilWatchCacheFreshForInitialEvents(ctx, requestedWatchRV, opts); err != nil {
+		return newErrWatcher(err), nil
+	}
+	startWatchRV := startWatchResourceVersionFn()
+	cacheInterval, err := c.watchCache.getAllEventsSinceLocked(startWatchRV)
```

Comment:
This still isn't correct, because for rv!=0, it will still return `initEvents` instead of the full state.

So what we want is for the `SendInitialEvents` case, we really need to force this function to return the full state.

Reply:
`initEvents` seems to be overloaded term here.

We need to return full state yes - these will be part of `initEvents`. But not every initEvent is a state...

Reply:
okay, I think i fixed the code. Now it reads all items from the store. PTAL.